tests/test_sampling.py

```diff
@@ -0,0 +1,18 @@
+import pytest
+from your_script_name import direct_sampling_with_times  # Replace with your actual script name
+import numpy as np
+
```

Comment:
## Test for Performance Constraints

import time

def test_performance():
    start_time = time.time()
    direct_sampling_with_times(Q, T)
    end_time = time.time()
    assert end_time - start_time < max_allowed_time